runtime-transaction/src/transaction_meta.rs

```diff
@@ -23,6 +23,7 @@ pub trait StaticMeta {
     fn is_simple_vote_transaction(&self) -> bool;
     fn signature_details(&self) -> &TransactionSignatureDetails;
     fn compute_budget_instruction_details(&self) -> &ComputeBudgetInstructionDetails;
+    fn instruction_data_len(&self) -> u64;
```

Comment:
what should be the relationship between `TransactionMeta` and `InstructionMeta`? In this case, `InstructionMeta` is internal and not visible to users of `TransactionMeta`, but adding a new instruction meta will require to updated both `InstructionMeta` and `TransactionMeta`; 

Or `TransactionMeta` can have `InstructionMeta`, make later visible to users, makes changes easier, but user needs to take one extra hop.

Reply:
I guess right now `InstructionMeta` is just an internal concept of all the instruction-based meta data we need to construct a `TransactionMeta`. It's just meant to be destructured and used to build the stored `TransactionMeta`